src/test/java/com/google/sps/MatchServletTest.java

```diff
@@ -41,27 +43,27 @@ public void setUp() throws Exception {
     }
 
     @Test
-    public void testMatchDoGet() throws IOException, ServletException {
+    public void doGet_returnsSavedMatches() throws IOException, ServletException {
         Gson gson = new Gson();
         Collection<Match> matches = testRepository.getMatchesForUser(testUser);
         String expected = gson.toJson(matches);
 
-        StringWriter sw = new StringWriter();
-        PrintWriter pw = new PrintWriter(sw);
-        when(response.getWriter()).thenReturn(pw);
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(printWriter);
 
         MatchServlet matchServlet = new MatchServlet();
         matchServlet.init();
         matchServlet.doGet(request, response);
 
-        String result = sw.getBuffer().toString().trim();
+        String result = stringWriter.getBuffer().toString().trim();
 
-        pw.flush();
-        assertEquals(expected, result);
+        printWriter.flush();
+        Assert.assertEquals(expected, result);
     }
 
     @Test
-    public void testMatchDoPost() throws IOException, ServletException {
+    public void doPost_storesNewMatches() throws IOException, ServletException {
         String[] matches = {"matchA", "matchB", "matchC"};
```

Comment:
It looks like this test verifies that the request.getParameterValues("new-matches") is called.

Would it make sense to use multiple requests to verify that new matches are stored? 
1. call doGet(), verify that the response has n matches
2. call doPost(), add another match for the user
4. call doGet(), verify that the response has n+1 matches

Let me know what you think

Reply:
I see what you are saying. I tried to implement this in the latest commit by testing calling doPost once and twice in two separate tests. Let me know if this is what you meant!

Reply:
I reworked the tests based on our conversation yesterday, let me know what you think